### DISCUSSION TITLE
Pyright does not respect a valid `# type: ignore` comment when a function arguments are on multiple lines.

### DISCUSSION THREAD

**Discussion by simon-liebehenschel** [microsoft/pyright#4358] [Q&A] [answered]
### Code example

Here is a very simplified code because I am talking not about type hints, but about how Pyright parses `# ignore` comments.

```python
@limit_gen
async def my_function(*, foo, bar):
    ...

# A decorator adds a `limit` parameter, and to suppress Mypy warnings during a function call I use `# type: ignore[call-arg]`

async for _ in my_function(  # type: ignore[call-arg]
    foo=bar,
    spam=eggs,
    limit=42,
):
...
```
### Mypy result

No errors because Mypy discovers a `# type: ignore[call-arg]` comment

### Pyright result

A warning, unless I will copy the comment to a specific argument line:
```python
my_function(  # type: ignore[call-arg]
    foo=bar,
    spam=eggs,
    limit=limit  # type: ignore[call-arg]
)
```

### Question

Is this a Pyright bug or a feature?

**Comment by erictraut** [comment, marked as answer]
A `# type: ignore` comment affects only the line that it is on. This is by design.

**Reply by iamb2** [reply]
@erictraut  how would i type ignore a multi-line statement ? 
here's what im trying to do
```
assert (  # type: ignore
            abs(1.0 - correlation) <= CUSTOM_TOLERANCES["correlation"]
        ), f"Correlation difference exceeds tolerance: {correlation}"
```

**Reply by erictraut** [reply]
Add the `# type: ignore` or `# pyright: ignore` to the end of the line where the error is reported. If the error is associated with a text range that spans lines, you can place it on any of the associated lines.
